src/components/NftPreview/NftPreview.tsx

```diff
@@ -127,10 +129,31 @@ function NftPreview({
     (event: React.MouseEvent<HTMLElement>) => {
       if (onClick && !event.metaKey && !event.ctrlKey && !event.shiftKey && !event.altKey) {
         event.preventDefault();
-        onClick();
+
+        // Replace the URL without triggering a page load
+        historyReplaceState(
+          route({
+            pathname: '/[username]/[collectionId]/[tokenId]',
+            query: {
+              username: username as string,
+              collectionId,
+              tokenId: token.dbid,
+              returnTo: 'home',
+            },
+          })
+        );
```

Comment:
any reason we're using replace? i think it's reasonable to assume that clicking "back" would just return the user to /home.

this is the behavior of a collection page as well:
1. visit collection: https://gallery.so/bastiboii/2GytPwuD9sxE2n3cd1yJdh2L2PC
2. click on NFT
3. click back
4. end up back on collection

Reply:
Oh right I forgot to mention why I used replace!

So, I'm not entirely sure why but when I use `pushState` the scroll position is lost as soon as you try to move it. So like everything works (the back button etc), but when you try to scroll away it just scrolls to the top.

I thought to have this as a somewhat temp solution until we fix this issue.

Reply:
ahhh i remember now. it's because on the main gallery page we use `LinkToNftDetailView` instead of imperatively pushing. 
- [component](https://github.com/gallery-so/gallery/blob/main/src/scenes/NftDetailPage/LinkToNftDetailView.tsx)
- [usage](https://github.com/gallery-so/gallery/blob/3dc745d80b407579db61e8f80ea6f68ed351c8ca/src/components/NftPreview/NftPreview.tsx#L204)
- make sure it's paired with [this hook](https://github.com/gallery-so/gallery/blob/3dc745d80b407579db61e8f80ea6f68ed351c8ca/src/scenes/NftDetailPage/useDisplayFullPageNftDetailModal.tsx#L12)
- [usage of the hook](https://github.com/gallery-so/gallery/blob/3dc745d80b407579db61e8f80ea6f68ed351c8ca/src/scenes/UserGalleryPage/UserGallery.tsx#L70)

does this make sense? i know it's a bit confusing. we should probably pair / co-locate the component + hook together